core/src/main/java/org/kiwix/kiwixmobile/core/utils/BookUtils.kt

```diff
@@ -15,46 +15,44 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-package org.kiwix.kiwixmobile.core.utils;
+package org.kiwix.kiwixmobile.core.utils
 
-import java.util.HashMap;
-import java.util.Locale;
-import java.util.Map;
+import java.util.HashMap
+import java.util.Locale
 
 /**
  * Created by mhutti1 on 19/04/17.
  */
-
-public class BookUtils {
-
-  public final Map<String, Locale> localeMap;
+class BookUtils {
+  val localeMap: Map<String, Locale>
 
   // Create a map of ISO 369-2 language codes
-  public BookUtils() {
-    String[] languages = Locale.getISOLanguages();
-    localeMap = new HashMap<>(languages.length);
-    for (String language : languages) {
-      Locale locale = new Locale(language);
-      localeMap.put(locale.getISO3Language(), locale);
-    }
+  init {
+    val languages = Locale.getISOLanguages()
+    localeMap = HashMap(languages.size)
+    languages
+      .asSequence()
+      .map(::Locale)
+      .forEach { localeMap.put(it.isO3Language, it) }
   }
 
   // Get the language from the language codes of the parsed xml stream
-  public String getLanguage(String languageCode) {
-
+  fun getLanguage(languageCode: String?): String {
+    var language = ""
```

Comment:
We were using `return` at the appropriate points in this function to avoid declaring a variable that made it more difficult to read.

This entire block could be expressed very well by using `when`